src/parser/core/metric.ts

```diff
@@ -1,11 +1,19 @@
-import { AnyEvent } from './Events';
+import { AnyEvent, CombatantInfoEvent } from './Events';
 import Ability from './modules/Ability';
 
 export interface Info {
+  /** The ID of the selected player */
   playerId: number;
+  /** The selected player's spellbook */
   abilities: Ability[];
+  /** The timestamp of the selected fight's start */
   fightStart: number;
+  /** The timestamp of the selected fight's end */
   fightEnd: number;
+  /** The combatantinfo for the selected player */
+  selectedCombatant?: CombatantInfoEvent;
```

Comment:
This should always be set, right?

Reply:
Yeah, but when I made it mandatory it broke your unit tests and my new unit tests because I had no good way of getting the combatant. What I probably actually need to do is make it mandatory and then mock the field out during testing.

Reply:
What unit tests did it break? I may have to do more Pick<>ing in my metrics.

Reply:
I guess the tests didn't break, they no longer type-check because the `Info` object created for testing doesn't have a selectedCombatant, and I couldn't figure out a quick way to mock one in. The solution is clearly to actually mock one in.

Reply:
I think it may be better to have these metrics only Pick the fields they use. That way we can actually see what tests need which fields and know what to test.

Reply:
Ah it's in this PR: https://github.com/WoWAnalyzer/WoWAnalyzer/pull/4562/files#diff-b4f839e317b6747bff68bd8b7fda426890d7b0c51c90e933259b74d3c985a89d